packages/next/build/webpack/plugins/wellknown-errors-plugin/parseRSC.ts

```diff
@@ -0,0 +1,90 @@
+import type { webpack } from 'next/dist/compiled/webpack/webpack'
+
+import { relative } from 'path'
+import { SimpleWebpackError } from './simpleWebpackError'
+
+export function formatRSCErrorMessage(
+  message: string
+): null | [string, string] {
+  if (message && /NEXT_RSC_ERR_/.test(message)) {
+    let formattedMessage = message
+    let formattedVerboseMessage = ''
+
+    // Comes from the "React Server Components" transform in SWC, always
+    // attach the module trace.
+    const NEXT_RSC_ERR_REACT_API = /.+NEXT_RSC_ERR_REACT_API: (.*?)\n/s
+    const NEXT_RSC_ERR_SERVER_IMPORT = /.+NEXT_RSC_ERR_SERVER_IMPORT: (.*?)\n/s
+    const NEXT_RSC_ERR_CLIENT_IMPORT = /.+NEXT_RSC_ERR_CLIENT_IMPORT: (.*?)\n/s
+
+    if (NEXT_RSC_ERR_REACT_API.test(message)) {
+      formattedMessage = message.replace(
+        NEXT_RSC_ERR_REACT_API,
+        `\n\nYou're importing a component that needs $1. It only works in a Client Component but none of its parents are marked with "client", so they're Server Components by default.\n\n`
+      )
+      formattedVerboseMessage =
+        '\n\nMaybe one of these should be marked as a "client" entry:\n'
+    } else if (NEXT_RSC_ERR_SERVER_IMPORT.test(message)) {
+      formattedMessage = message.replace(
+        NEXT_RSC_ERR_SERVER_IMPORT,
+        `\n\nYou're importing a component that imports $1. It only works in a Client Component but none of its parents are marked with "client", so they're Server Components by default.\n\n`
+      )
+      formattedVerboseMessage =
+        '\n\nMaybe one of these should be marked as a "client" entry:\n'
+    } else if (NEXT_RSC_ERR_CLIENT_IMPORT.test(message)) {
+      formattedMessage = message.replace(
+        NEXT_RSC_ERR_CLIENT_IMPORT,
+        `\n\nYou're importing a component that needs $1. That only works in a Server Component but one of its parents is marked with "client", so it's a Client Component.\n\n`
```

Comment:
Should we add `err.sh` docs for these so that we can show examples of this better?

Reply:
Can `err.sh` support custom query parameters (like React's error) to render, or 100% static? Because the `$1` here is dynamic (there are many possible APIs that will cause an error here, and we might extend).

I can add some static content there as well to explain it better (or maybe linking to the RSC docs?)

Reply:
Oh it doesn't need to be dynamic just meant something static like https://github.com/vercel/next.js/blob/canary/errors/no-head-element.md where we can add related docs links and such